pkg/api/kyverno/v1/types.go

```diff
@@ -120,34 +128,52 @@ type PolicyViolationList struct {
 type Policy struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              Spec         `json:"spec"`
-	Status            PolicyStatus `json:"status,omitempty"`
+	// Spec is the information to identify the policy
+	Spec Spec `json:"spec"`
+	// Status contains statistics related to policy
+	Status PolicyStatus `json:"status,omitempty"`
 }
 
 // Spec describes policy behavior by its rules
 type Spec struct {
-	Rules                   []Rule `json:"rules"`
+	// Rules contains the list of rules to be applied to resources
+	Rules []Rule `json:"rules"`
+	// ValidationFailureAction provides choice to enforce rules to resources during policy violations.
+	// Default value is "audit".
 	ValidationFailureAction string `json:"validationFailureAction"`
-	Background              *bool  `json:"background"`
+	// Background provides choice for applying rules to existing resources.
+	// Default value is "true".
+	Background *bool `json:"background"`
 }
 
 // Rule is set of mutation, validation and generation actions
 // for the single resource description
 type Rule struct {
-	Name             string           `json:"name"`
-	MatchResources   MatchResources   `json:"match"`
+	// Name - A required field represents rule name
+	Name string `json:"name"`
+	// (Optional): Specifies resources for which the rule has to be applied.
+	// If it's defined, "kind" inside MatchResources block is required.
+	MatchResources MatchResources `json:"match"`
```

Comment:
Rather then adding comment can we add comment like this `// +optional`

```
//Specifies resources for which the rule has to be applied.
// If it's defined, "kind" inside MatchResources block is required.
// +optional
MatchResources MatchResources `json:"match"`
```
Check generated output here https://htmlpreview.github.io/?https://github.com/kubernetes-sigs/wg-policy-prototypes/blob/master/policy-report/docs/index.html#policy.kubernetes.io/v1alpha1.*github.com/kubernetes-sigs/wg-policy-prototypes/policy-report/api/v1alpha1.PolicyReportResult

Reply:
I'll check that @evalsocket .